packages/gateway/src/resolvers/domain.ts

```diff
@@ -0,0 +1,62 @@
+import { Hex, labelhash, namehash, zeroAddress } from 'viem'
+
+import { Text, Address, DomainMetadata, NodeProps } from '../types'
+import { Domain } from '../entities'
+
+interface ReadRepository {
+  getDomain(params: NodeProps): Promise<Domain | null>
+  getSubdomains({ node }: NodeProps): Promise<string[]>
+  getTexts({ node }: NodeProps): Promise<Text[]>
+  getAddresses({ node }: NodeProps): Promise<Address[]>
+}
+
+interface Client {
+  getOwner(node: Hex): Promise<Hex>
+  getResolver(node: Hex): Promise<Hex | undefined>
+}
+
+export async function domainResolver(
+  { name }: { name: string },
+  repo: ReadRepository,
+  client: Client,
+): Promise<DomainMetadata | undefined> {
+  const node = namehash(name)
+  const domain = await repo.getDomain({ node })
+  const resolver = domain?.resolver || (await client.getResolver(node))
+  if (!resolver || resolver === zeroAddress) return
+
+  const [, labelName] = /(.*)\.eth/.exec(name) || []
```

Comment:
This should be the last level of the domain.

Eg.: sid.cb.id ; label is sid, parent is cb.id